geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/RedisCommandType.java

```diff
@@ -154,207 +165,274 @@
-  SUNIONSTORE(new SUnionStoreExecutor(), UNSUPPORTED, new MinimumParameterRequirements(3)),
+  SCARD(new SCardExecutor(), UNSUPPORTED, new Parameter().exact(2).flags(readonly, fast)),
+  SDIFF(new SDiffExecutor(), UNSUPPORTED,
+      new Parameter().min(2).lastKey(-1).flags(readonly, sort_for_script)),
+  SDIFFSTORE(new SDiffStoreExecutor(), UNSUPPORTED,
+      new Parameter().min(3).lastKey(-1).flags(write, denyoom)),
+  SINTER(new SInterExecutor(), UNSUPPORTED,
+      new Parameter().min(2).lastKey(-1).flags(readonly, sort_for_script)),
+  SINTERSTORE(new SInterStoreExecutor(), UNSUPPORTED,
+      new Parameter().min(3).lastKey(-1).flags(write, denyoom)),
+  SISMEMBER(new SIsMemberExecutor(), UNSUPPORTED, new Parameter().exact(3).flags(readonly, fast)),
+  SMOVE(new SMoveExecutor(), UNSUPPORTED, new Parameter().exact(4).lastKey(2).flags(write, fast)),
+  SPOP(new SPopExecutor(), UNSUPPORTED, new Parameter().min(2).max(3, ERROR_SYNTAX)
+      .custom(SpopParameterRequirements.checkParameters()).flags(write, random, fast)),
+  SRANDMEMBER(new SRandMemberExecutor(), UNSUPPORTED,
+      new Parameter().min(2).flags(readonly, random)),
+  SSCAN(new SScanExecutor(), UNSUPPORTED, new Parameter().min(3).flags(readonly, random),
+      new Parameter().odd(ERROR_SYNTAX).firstKey(0)),
+  SUNION(new SUnionExecutor(), UNSUPPORTED,
+      new Parameter().min(2).lastKey(-1).flags(readonly, sort_for_script)),
+  SUNIONSTORE(new SUnionStoreExecutor(), UNSUPPORTED,
+      new Parameter().min(3).lastKey(-1).flags(write, denyoom)),
 
   /*************** Server ****************/
 
-  DBSIZE(new DBSizeExecutor(), UNSUPPORTED, new ExactParameterRequirements(1)),
-  FLUSHALL(new FlushAllExecutor(), UNSUPPORTED, new MaximumParameterRequirements(2, ERROR_SYNTAX)),
-  FLUSHDB(new FlushAllExecutor(), UNSUPPORTED, new MaximumParameterRequirements(2, ERROR_SYNTAX)),
-  SHUTDOWN(new ShutDownExecutor(), UNSUPPORTED, new MaximumParameterRequirements(2, ERROR_SYNTAX)),
-  TIME(new TimeExecutor(), UNSUPPORTED, new ExactParameterRequirements(1)),
+  DBSIZE(new DBSizeExecutor(), UNSUPPORTED, new Parameter().exact(1).firstKey(0).flags(readonly, fast)),
+  FLUSHALL(new FlushAllExecutor(), UNSUPPORTED,
+      new Parameter().min(1).max(2, ERROR_SYNTAX).firstKey(0).flags(write)),
+  FLUSHDB(new FlushAllExecutor(), UNSUPPORTED,
+      new Parameter().min(1).max(2, ERROR_SYNTAX).firstKey(0).flags(write)),
+  SHUTDOWN(new ShutDownExecutor(), UNSUPPORTED,
+      new Parameter().min(1).max(2, ERROR_SYNTAX).firstKey(0).flags(admin, noscript, loading,
+          stale)),
+  TIME(new TimeExecutor(), UNSUPPORTED,
+      new Parameter().exact(1).firstKey(0).flags(random, loading, stale, fast)),
 
   /***************************************
    *********** Unknown Commands **********
    ***************************************/
   UNKNOWN(new UnknownExecutor(), RedisCommandSupportLevel.UNKNOWN);
 
+  public enum Flag {
+    admin,
```

Comment:
I think our style guide for enum members is that they are uppercase since they are constants. These are all lowercase. Also you have some that use an underscore but others that are multiple words (denyoom, movablekeys, noscript, readonly) but do not use an underscore.

Reply:
I've made them all uppercase now. The reason some use underscores is to match the flags in redis - https://redis.io/commands/command and https://redis.io/topics/modules-api-ref